src/utils.py

```diff
@@ -830,6 +830,21 @@ def extract_preds_and_types(
     return preds, types
 
 
+def get_static_preds(ground_ops: Collection[GroundNSRTOrSTRIPSOperator],
```

Comment:
so this utility is useful when we don't have just one initial state, is that the idea?

that makes sense. can we actually then do this on the basis of lifted NSRTs/operators instead of ground ones, to avoid the potentially expensive grounding? I believe we'll be able to identify the same predicates from the lifted operators alone, but let me know if I'm wrong.

Reply:
oh but actually,, we have a `state` here. so why do we need another utility?

Reply:
oh i did not see this. hmmm

Reply:
wait so do we want to avoid the grounding or should I just use the existing utility?

Reply:
i'll try to do a light refactor that might help here. will let you know if it works...

Reply:
see #320

Reply:
>wait so do we want to avoid the grounding or should I just use the existing utility?
let's use the existing utility for now, unless grounding takes more than ~1 second, in which case we can use rohan's adaption of your new utility

Reply:
why not just use the utility i just made? i made it for this reason lol

Reply:
because the one that uses ground nsrts is strictly better, efficiency aside, I think. it will be able to filter out more static things.

Reply:
you just have to use the utils.py from master, and then Amber's change to src/approaches/interactive_learning_approach.py should be all you need

Reply:
@amburger66 sorry for the back and forth here, rohan and I touched base and we're on the same page now. so here's what we think:
1. if grounding is not a noticeable slowdown, then you can use `utils.get_static_atoms`. in this case, you will need to refactor `glib_sample`, perhaps doing the static atoms calculation inside of glib_sample instead (because right now you're just filtering out predicates rather than static atoms).
2. if grounding is causing a slowdown, then you can use `utils.get_static_predicates`. here you wouldn't need to change the `glib_sample` code.

let me know if that's clear. thanks again!

Reply:
(minor caveat: we realized we actually just made a bug in the new `utils.get_static_atoms`, which we'll fix shortly in a new PR 😬 )

Reply:
fixing that bug in #321. science is hard!

Reply:
makes sense!

Reply:
keeping it as is since grounding is ~2.5s slower (playroom has maaaany static atoms)